### DISCUSSION TITLE
omitting libraries to speed up build times during firmware development?

### DISCUSSION THREAD

**Discussion by efm-7** [meshtastic/firmware#4861] [Q&A]
Good evening, this is my first ever post on github and coding is not exactly my favorite activity (despite it somewhat being my career, in a rather closed environment mind you) vacuum tubes, analog, and linear circuitry is what makes me happy, but i'm giving this a shot! 

I'm building my own firmware variant based on the Heltec e290, however it's taking like 11 minutes to build  every time and I don't think I have patience for that given that I want to do a lot of back to back trial and error. 

**Are there any low hanging fruits for things I can omit/comment out to speed up the build times just while i'm developing?** 

I'm not using any external sensors nor am I using any radios other than the SX1262... 

EDIT: I have switched to using a Heltec T190 as a base, for unrelated issues I was having, build times are still hovering around 7-8 min which is still not ideal

**Comment by fifieldt** [comment]
welcome, it's great to have you!

You can try adding these to your variant.h:

#define MESHTASTIC_EXCLUDE_GPS

#define MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR

**Reply by efm-7** [reply]
Thank you! I will give this a shot!

**Reply by efm-7** [reply]
Unfortunately, it didn't quite work out :(
I'm not sure if those are being overridden, but I did find this in configuration.h which was a bit of a clue

#define MESHTASTIC_MINIMIZE_BUILD 1

However, defining this will pretty much turn off everything, and that's not what I want. So I have this in my variant.h but it doesn't seem to be excluding much if anything at all. build times are still the same.

#define MESHTASTIC_EXCLUDE_WIFI 1
#define MESHTASTIC_EXCLUDE_GPS 1
#define MESHTASTIC_EXCLUDE_MQTT 1
#define MESHTASTIC_EXCLUDE_TZ 1
#define MESHTASTIC_EXCLUDE_AUDIO 1
#define MESHTASTIC_EXCLUDE_DETECTIONSENSOR 1
#define MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR 1
#define MESHTASTIC_EXCLUDE_PAXCOUNTER 1
#define MESHTASTIC_EXCLUDE_RANGETEST 1
#define MESHTASTIC_EXCLUDE_ATAK 1
#define MESHTASTIC_EXCLUDE_CANNEDMESSAGES 1
#define MESHTASTIC_EXCLUDE_INPUTBROKER 1

**Reply by fifieldt** [reply]
Perhaps have a look in platformio.ini and see what libraries you can delete from there.
